### PR TITLE
[Experimental] Allow overriding placeholder in interactivity Dropdown::render

### LINKED ISSUE #43205: Add support to interactivity Dropdown.php to override default placeholder
Right now the default placeholder rendered by the interactivity Dropdown is not customizable. This task involves allowing a prop to be passed to `Dropdown::render` to override the placeholder.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

In my [previous PR](https://github.com/woocommerce/woocommerce/pull/43183#discussion_r1438998009) @dinhtungdu mentioned it would be good to be able to override the placeholder text provided to the Dropdown.php interactivity component.

This provides that functionality. We now determine the default placeholder text once from in PHP instead of front-end (and pass it as context to front-end) and if you want to override it you can pass it as a prop of `placeholder`.

This also adjusts the attribute filter to work with it and fixes a couple of bugs I noticed in the filter when I was working with it in Dropdown mode.

Also note there is a behaviour I'm not sure should be correct which is that the attributes available to choose from in the dropdown get filtered down to those available in the currently filtered products. I'm not sure that's desirable behavior for this kind of filter. wdyt @dinhtungdu ?

Closes #43205

### How to test the changes in this Pull Request:

1. Create a collection filters block
2. Select the attribute filter and in the tools, choose "dropdown" for the mode.
3. Preview in front-end and see that the placeholder matches the attribute name, and is kept up to date if you select then deselect an attribute from the dropdown.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
[Experimental] support passing a placeholder text to the interactivity Dropdown component.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/collection-filters/inner-blocks/attribute-filter/frontend.ts [hunk 1]
```diff
@@ -50,10 +50,10 @@ store( 'woocommerce/collection-attribute-filter', {
 
 			const context = getContext< AttributeFilterContext >();
 
-			if ( dropdownContext.selectedItem.value ) {
+			if ( dropdownContext.selectedItems[ 0 ].value ) {
 				navigate(
 					getUrl(
-						[ dropdownContext.selectedItem.value ],
+						[ dropdownContext.selectedItems[ 0 ].value ],
```
Comment (samueljseay) [inline plugins/woocommerce-blocks/assets/js/blocks/collection-filters/inner-blocks/attribute-filter/frontend.ts]: Looks like this bug was hanging around from when I converted dropdown to support multiple items.

File: plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php [hunk 2]
```diff
@@ -231,7 +233,9 @@ private function render_attribute_dropdown( $options, $attributes ) {
 			array(
 				'items'          => $list_items,
 				'action'         => 'woocommerce/collection-attribute-filter::actions.navigate',
-				'selected_items' => array( $selected_item ),
+				'selected_items' => empty( $selected_item ) ? array() : array( $selected_item ),
+				// translators: %s is a product attribute name.
+				'placeholder'    => sprintf( __( 'Select %s', 'woocommerce' ), $product_attribute->name ),
```
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/BlockTypes/CollectionAttributeFilter.php:239]: I think we should also do the same for rating and stock filter blocks.

**Review by dinhtungdu — APPROVED** [review]
Apart from the review comment, this is LGTM!

**Comment by dinhtungdu** [comment]
@samueljseay Before https://github.com/woocommerce/woocommerce/pull/43211/commits/a236a4f42f666ff64a27bc53c169c6a3b8749ef1, active filter can't be removed from the Dropdown (clicking on the remove button of the selected items doesn't do anything).

**Review by dinhtungdu — APPROVED** [review]
This is LGTM now! Thank you so much!

**Pull request merged**